### PR TITLE
fix: --show-diff missing on chart version bump

### LINKED ISSUE #555: -show-diff missing on chart version bump
Hi @luisdavim, looks like these change prevent the diff from printing when using the -show-diff flag with a chart version bump, as it shortcuts the diff test: https://github.com/Praqma/helmsman/blob/master/internal/app/decision_maker.go#L323

Maybe r.diff() should run in all cases with -show-diff or -verbose mode?

### PR DESCRIPTION
**Pull request by luisdavim**
This should fix #555

### CODE REVIEW

File: internal/app/release.go [hunk 1]
```diff
@@ -401,6 +397,9 @@ func (r *release) getHelmArgsFor(action string, optionalNamespaceOverride ...str
 }
 
 func (r *release) checkChartDepUpdate() {
+	if r == nil || r.disabled {
```
Comment (antoinedeschenes) [inline internal/app/release.go:400]: Could probably use `!r.isConsideredToRun` here

File: internal/app/decision_maker.go [hunk 2]
```diff
@@ -305,31 +308,43 @@ func (cs *currentState) inspectUpgradeScenario(r *release, p *plan, c *chartInfo
...
 		r.upgrade(p)
 		p.addDecision("Release [ "+r.Name+" ] will be upgraded", r.Priority, change)
-		return
+		return nil
 	}
-	if diff := r.diff(); diff != "" {
+
+	if diff, err := r.diff(); err != nil {
+		return err
+	} else if diff != "" {
```
Comment (antoinedeschenes) [inline internal/app/decision_maker.go:342]: We're missing  `if flags.verbose || flags.showDiff {fmt.Println(diff)}` here

File: internal/app/release.go (hunk 1 above)
Comment (luisdavim) [reply internal/app/release.go:400]: sure.

**Comment by antoinedeschenes** [comment]
lgtm, show-diff on update and upgrade works properly

**Review by antoinedeschenes — APPROVED** [review]

**Pull request merged**